src/test/java/hudson/plugins/mercurial/ConfigurationTest.java

```diff
@@ -91,6 +92,17 @@
         assertFalse(scm.isDisableChangeLog());
     }
 
+    @Test public void doFillCredentialsIdItemsWithoutJobWhenAdmin() throws Exception {
+        r.jenkins.setSecurityRealm(r.createDummySecurityRealm());
+        MockAuthorizationStrategy as = new MockAuthorizationStrategy();
+        // This AuthorizationStrategy has the ADMINISTER permission granted by default
+        r.jenkins.setAuthorizationStrategy(as);
+        UsernamePasswordCredentialsImpl c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, null, "test", "bob", "s3cr3t");
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+        ListBoxModel options = r.jenkins.getDescriptorByType(MercurialSCM.DescriptorImpl.class).doFillCredentialsIdItems(null, "http://nowhere.net/");
```

Comment:
You need to use `ACL.impersonate`. Otherwise you are checking as `SYSTEM`, bypassing *all* permission checks.